app/frontend/src/cljs/teet/land/owner_opinion_view.cljs

```diff
@@ -163,18 +225,185 @@
+                           response-date :land-owner-opinion/date
+                           link-to-response :land-owner-opinion/link-to-response
+                           creator :meta/creator
+                           modifier :meta/modifier :as opinion}
+                       edit-right? editing?]
+   [:div {:id (str "opinion-" id)}
+    (when (not editing?)
+      [Grid {:container true
+             :spacing 2}
+       [Grid {:item true
+              :md 6
+              :xs 12}
+        [typography/TextBold (tr [:fields :land-owner-opinion/date])]]
+       [Grid {:item true
+              :md 6
+              :xs 12}
+        [typography/SmallText (tr [:fields :land-owner-opinion/link-to-response])]]
+       [Grid {:item true
+              :md 6
+              :xs 12}
+        [typography/TextBold (format/date response-date)]]
+       [Grid {:item true
+              :md 6
+              :xs 12}
+        [typography/SmallText [:a {:target :_blank
+                                   :href link-to-response} link-to-response]]]
+       [Grid {:item true
+              :md 6
+              :xs 12}
+        [typography/Heading4 (tr [:fields :land-owner-opinion/body])]]
+       [Grid {:item true
+              :md 6
+              :xs 12}
+        [Grid {:container true
+               :spacing 0
+               :direction :row
+               :align :flex-end}
+         [Grid {:item true
+                :md 6
+                :xs 12}
+          [typography/Heading4 (tr [:fields :land-owner-opinion/authority-position])]]
+         [Grid {:item true
+                :md 6
+                :xs 12}
+          [typography/Heading5 (if modifier
+                                      (str (:user/given-name modifier) " " (:user/family-name modifier))
+                                      (str (:user/given-name creator) " " (:user/family-name creator)))]]]]
+       [Grid {:item true
+              :md 6
+              :xs 12}
+        [rich-text-editor/display-markdown body]]
+       [Grid {:item true
+              :md 6
+              :xs 12}
+        [rich-text-editor/display-markdown authority-position]]])])
+
+(defn- get-opinion-data-for-update
+  "Select updatable data from opinion and transform activity enum to key word to be selectable"
+  [opinion]
+  (->
+    (select-keys opinion
+      [:db/id :land-owner-opinion/activity :land-owner-opinion/body
+       :land-owner-opinion/type :land-owner-opinion/date
+       :land-owner-opinion/respondent-connection-to-land
+       :land-owner-opinion/authority-position :land-owner-opinion/respondent-name
+       :land-owner-opinion/link-to-response
+       :meta/creator :meta/modifier])
+    (update-in [:land-owner-opinion/activity :activity/name] du/enum->kw)))
+
+(defn owner-opinion-details
+  "Contains Opinion details update form and \"Edit\" button to show or hide it"
+  [e! {:keys [edit-rights?]} {id :db/id :as opinion} project target refresh!]
+  (r/with-let [[pfrom pto] (common/portal)
```

Comment:
maybe could evaluate if this portal business is necessary here since the html structure seems like the portals would be unnecessary.

Reply:
As we agreed after discussion in Slack as the refactoring is bigger than initially thought we keep the portal for now.